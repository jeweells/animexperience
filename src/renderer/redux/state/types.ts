import { FStatus, Optional, TopView } from '@shared/types'
import {
  AnimeInfo,
  DeepAnimeIdSearchResult,
  EpisodeInfo,
  FollowedAnime,
  RecentAnimeInfo,
  RecommendationInfo,
  WatchHistoryItem
} from '@shared/types'
import { FollowedAnimeWStatus } from '../reducers/followedAnimes'
import { PreAllowWatch } from '../reducers/invokedLink'
import { RecentAnimeData } from '../../src/hooks/useRecentAnimes'
import { VideoOption } from '../../src/components/VideoPlayer'

export type AnimeSearchState = {
  // Boolean id
  searching?: string
  result?: Optional<DeepAnimeIdSearchResult>
  status: Partial<{
    result: FStatus
    moreResults: FStatus
  }>
}

export type FollowedAnimesState = {
  followedDict: Record<string, FollowedAnime>
  followed: FollowedAnimeWStatus[]
  status: Partial<{
    followed: FStatus
  }>
}

export type InvokedLinkState = {
  open: {
    watch?: string
  }
  preAllow: Partial<{
    watch: PreAllowWatch
  }>
  status: {
    watch?: FStatus
  }
}

export type NotificationsState = {
  message: string | null
  key: string
}

export type PeekState = {
  name?: string
  status: Partial<{
    info: FStatus
  }>
  info?: Optional<AnimeInfo>
  peeking?: string
}

export type PlayerState = {
  open?: string
  freezed?: boolean
}

export type OptionInfo = {
  name: string
  prefer?: boolean
  score: number
}
// Define a type for the slice state
export type PlayerOptionsState = {
  options?: OptionInfo[]
  history?: string[]
  preferred?: string[]
  status: Partial<{
    options: FStatus
  }>
}

export type RecommendationsState = {
  [name: string]: Partial<{
    status: FStatus
    recommendations: RecommendationInfo[]
  }>
}

export type TopviewState = {
  views: TopView[]
}

export type WatchState = {
  watching?: Optional<RecentAnimeData>
  availableVideos?: Optional<VideoOption[]>
  status: {
    availableVideos?: FStatus
    info?: FStatus
    watchEpisode?: FStatus
  }
  info?: Optional<AnimeInfo>
  showNextEpisodeButton?: boolean
  nextEpisodeTimeout: number
  autoFullScreen?: boolean
}

export type WatchedState = {
  episodes: {
    [name: string]: EpisodeInfo
  }
  recently?: RecentAnimeInfo[]
  status: {
    episodes?: FStatus
    recently?: FStatus
  }
}

export type WatchHistoryState = {
  sorted?: WatchHistoryItem[]
  status: {
    sorted?: FStatus
  }
}
