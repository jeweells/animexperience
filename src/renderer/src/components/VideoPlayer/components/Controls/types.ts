export type ControlsContext = {
  video: HTMLVideoElement | null
}
