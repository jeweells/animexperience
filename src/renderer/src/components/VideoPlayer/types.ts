import { RecentAnimeData } from '../../hooks/useRecentAnimes'
import { Optional } from '@shared/types'
import { VideoOption } from './index'

export type $IframeContents = JQuery<HTMLIFrameElement | Text | Comment | Document>
export type BasicVideoInfo = {
  option?: Optional<VideoOption>
  anime?: Optional<RecentAnimeData>
}

export type KnownOption =
  | 'mega'
  | 'fembed'
  | 'okru'
  | 'mixdrop'
  | 'streamtape'
  | 'stape'
  | 'maru'
  | 'sw'
  | 'yourupload'
