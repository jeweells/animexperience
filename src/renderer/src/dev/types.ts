import { RawDevMessage } from '@shared/types'
import { Moment } from 'moment'

export type DevMessage = RawDevMessage & {
  id: string
  at: Moment
}
