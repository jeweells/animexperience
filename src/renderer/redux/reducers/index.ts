export { animeSearch } from './animeSearch'
export { followedAnimes } from './followedAnimes'
export { invokedLink } from './invokedLink'
export { notifications } from './notifications'
export { peek } from './peek'
export { player } from './player'
export { playerOptions } from './playerOptions'
export { recommendations } from './recommendations'
export { topView } from './topView'
export { watch } from './watch'
export { watched } from './watched'
export { watchHistory } from './watchHistory'
