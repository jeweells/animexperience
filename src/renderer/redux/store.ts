import { configureStore } from '@reduxjs/toolkit'
import animeSearch from './reducers/animeSearch'
import followedAnimes from './reducers/followedAnimes'
import peek from './reducers/peek'
import player from './reducers/player'
import playerOptions from './reducers/playerOptions'
import recommendations from './reducers/recommendations'
import topView from './reducers/topView'
import watch from './reducers/watch'
import watched from './reducers/watched'
import watchHistory from './reducers/watchHistory'
import invokedLink from './reducers/invokedLink'
import notifications from './reducers/notifications'

export default configureStore({
  reducer: {
    watch,
    player,
    watched,
    recommendations,
    watchHistory,
    peek,
    topView,
    playerOptions,
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications
  }
})
