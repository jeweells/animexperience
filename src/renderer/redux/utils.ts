import { ThunkDispatch } from 'redux-thunk'
import { RootState } from './state'
import { AnyAction } from '@reduxjs/toolkit'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { ForcedAny } from '@shared/types'

export type AppDispatch = ThunkDispatch<RootState, ForcedAny, AnyAction>

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export const useAppDispatch = useDispatch as () => AppDispatch
