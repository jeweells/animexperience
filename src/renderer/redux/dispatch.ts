import { AppDispatch } from '~/redux/utils'
import store from '~/redux/store'

export const dispatch: AppDispatch = (action) => store.dispatch(action)
