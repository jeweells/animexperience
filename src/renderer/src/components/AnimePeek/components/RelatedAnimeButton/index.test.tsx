import { Provider } from 'react-redux'
import RelatedAnimeButton, { RelatedAnimeButtonProps, types } from './index'
import { RootState } from '~/redux/state'
import { DeepPartial } from 'redux'
import { MockStoreEnhanced } from 'redux-mock-store'
import TopLayout from '../../../../../plugins/gatsby-plugin-top-layout/TopLayout'
import { RelatedAnime } from '@shared/types'
import { RELATED_ANIME_BUTTON } from '@selectors'

describe('RelatedAnimeButton', () => {
  let initialState: DeepPartial<RootState>
  let props: RelatedAnimeButtonProps
  let store: MockStoreEnhanced

  const getComponent = (override: Partial<RelatedAnimeButtonProps> = {}) => {
    store = mockStore(initialState)
    return (
      <TopLayout>
        <Provider store={store}>
          <RelatedAnimeButton {...props} {...override} />
        </Provider>
      </TopLayout>
    )
  }

  beforeEach(() => {
    initialState = {
      watched: {
        episodes: {
          69: {
            currentTime: 59,
            at: 101,
            duration: 600
          }
        }
      }
    }

    props = {
      related: {
        name: 'Related Anime Name',
        link: 'here goes related anime link',
        type: 'Ova'
      }
    }
  })

  it.each(types.map((type) => [type]))('renders default type=%s', (type) => {
    const wrapper = render(
      getComponent({
        related: { ...props.related, type: type as RelatedAnime['type'] }
      })
    )
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('clicks the button', () => {
    const wrapper = render(getComponent())
    fireEvent.click(wrapper.getByTestId(RELATED_ANIME_BUTTON.BUTTON))
    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        meta: expect.objectContaining({
          arg: props.related.name
        }),
        type: 'peek/peek/pending'
      })
    )
  })
})
