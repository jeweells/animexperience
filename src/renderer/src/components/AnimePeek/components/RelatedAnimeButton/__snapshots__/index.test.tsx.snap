// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedAnimeButton renders default type=Especial 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
  height: 36px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0 .TypeButton {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: rgb(160 99 70);
}

.emotion-0 .NBWrapper {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #292d33;
}

.emotion-0:hover .TypeButton {
  background-color: rgb(179, NaN, NaN);
}

.emotion-0:hover .NBWrapper {
  background-color: #3c3f43;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 12px;
  font-size: 0.85rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-width: 9ch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  padding: 0 12px;
}

.emotion-3 {
  margin: 0;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  text-align: left;
  font-size: 0.85rem;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<button
    btype="Especial"
    class="MuiButtonBase-root emotion-0"
    data-testid="related-anime-button-button"
    tabindex="0"
    title="Related Anime Name"
    type="button"
  >
    <div
      class="emotion-1 TypeButton"
    >
      Especial
    </div>
    <div
      class="emotion-2 NBWrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-3"
      >
        Related Anime Name
      </p>
    </div>
    <span
      class="MuiTouchRipple-root emotion-4"
    />
  </button>
</DocumentFragment>
`;

exports[`RelatedAnimeButton renders default type=Ova 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
  height: 36px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0 .TypeButton {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #612f2f;
}

.emotion-0 .NBWrapper {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #292d33;
}

.emotion-0:hover .TypeButton {
  background-color: rgb(128, 88, 88);
}

.emotion-0:hover .NBWrapper {
  background-color: #3c3f43;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 12px;
  font-size: 0.85rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-width: 9ch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  padding: 0 12px;
}

.emotion-3 {
  margin: 0;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  text-align: left;
  font-size: 0.85rem;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<button
    btype="Ova"
    class="MuiButtonBase-root emotion-0"
    data-testid="related-anime-button-button"
    tabindex="0"
    title="Related Anime Name"
    type="button"
  >
    <div
      class="emotion-1 TypeButton"
    >
      Ova
    </div>
    <div
      class="emotion-2 NBWrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-3"
      >
        Related Anime Name
      </p>
    </div>
    <span
      class="MuiTouchRipple-root emotion-4"
    />
  </button>
</DocumentFragment>
`;

exports[`RelatedAnimeButton renders default type=Pel√≠cula 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
  height: 36px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0 .TypeButton {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: rgb(55 92 95);
}

.emotion-0 .NBWrapper {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #292d33;
}

.emotion-0:hover .TypeButton {
  background-color: rgb(95, NaN, NaN);
}

.emotion-0:hover .NBWrapper {
  background-color: #3c3f43;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 12px;
  font-size: 0.85rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-width: 9ch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  padding: 0 12px;
}

.emotion-3 {
  margin: 0;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  text-align: left;
  font-size: 0.85rem;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<button
    btype="Pel√≠cula"
    class="MuiButtonBase-root emotion-0"
    data-testid="related-anime-button-button"
    tabindex="0"
    title="Related Anime Name"
    type="button"
  >
    <div
      class="emotion-1 TypeButton"
    >
      Pel√≠cula
    </div>
    <div
      class="emotion-2 NBWrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-3"
      >
        Related Anime Name
      </p>
    </div>
    <span
      class="MuiTouchRipple-root emotion-4"
    />
  </button>
</DocumentFragment>
`;

exports[`RelatedAnimeButton renders default type=Serie 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  border-radius: 4px;
  overflow: hidden;
  width: 100%;
  height: 36px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0 .TypeButton {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #427cb3;
}

.emotion-0 .NBWrapper {
  -webkit-transition: background-color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out;
  background-color: #292d33;
}

.emotion-0:hover .TypeButton {
  background-color: rgb(103, 150, 194);
}

.emotion-0:hover .NBWrapper {
  background-color: #3c3f43;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 12px;
  font-size: 0.85rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-width: 9ch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
  padding: 0 12px;
}

.emotion-3 {
  margin: 0;
  font-family: OpenSans,Quicksand,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  text-align: left;
  font-size: 0.85rem;
}

.emotion-4 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<button
    btype="Serie"
    class="MuiButtonBase-root emotion-0"
    data-testid="related-anime-button-button"
    tabindex="0"
    title="Related Anime Name"
    type="button"
  >
    <div
      class="emotion-1 TypeButton"
    >
      Serie
    </div>
    <div
      class="emotion-2 NBWrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 emotion-3"
      >
        Related Anime Name
      </p>
    </div>
    <span
      class="MuiTouchRipple-root emotion-4"
    />
  </button>
</DocumentFragment>
`;
